### PR TITLE
Gui cleanup

### PR DESCRIPTION
**Pull request by johnbotris**
Cleanup of some old components as part of #2559

See commits for specifics

Admin client has been adapted on a branch of the same name

### CODE REVIEW

File: src/gui/base/RichTextToolbar.ts [hunk 1]
```diff
@@ -1,190 +1,195 @@
...
+		animateToolbar(dom, true)
+	}
+
+	onbeforeremove(vnode: VnodeDOM<any>): Promise<void> {
+		return animateToolbar(vnode.dom as HTMLElement, false)
+	}
+
+	view({attrs}: Vnode<RichTextToolbarAttrs>): Children {
+
+		try {
```
Comment (charlag) [inline src/gui/base/RichTextToolbar.ts:38]: hm probably parsing it each `view()` is not a great idea, probably the easiest is to memoize it?

**Review by charlag — COMMENTED** [review]
LGTM, a lot of work

Tests?
Test notes?

File: src/gui/base/Label.ts [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
 import type {TranslationKey} from "../../misc/LanguageViewModel"
+import {lang} from "../../misc/LanguageViewModel"
 import type {lazy} from "@tutao/tutanota-utils"
 
 export type LabelAttrs = {
 	label: TranslationKey | lazy<string>
 }
 
-export class LabelN implements Component<LabelAttrs> {
+export class Label implements Component<LabelAttrs> {
```
[the remark this answers is not in the document]
Comment (johnbotris) [reply src/gui/base/Label.ts:12]: :detective:

**Comment by johnbotris** [comment]
# Test notes

Test these GUI components which have been modified
- [ ] Buttons in ContactFormEditor
- [ ] Buttons in ContactFormRequestDialog
- [ ] Buttons in ContactView
- [ ] Buttons in UserViewer
- [ ] Table in UserViewer
- [ ] Buttons in SubscriptionViewer
- [ ] Buttons in MultiSelectionBar
- [ ] DropdownSelector buttons
- [ ] General Button (check a few here and there)
- [ ] Table (check it in the settings viewers)
- [ ] Test the 404 Not Found page
- [ ] DropdownSelector in GlobalSettingsView
- [ ] DropdownSelector in EditSignatureDialog
- [ ] DropdownSelector in ContactFormEditor
- [ ] DropdownSelector in HtmlEditor
- [ ] DropdownSelector in PaymentDataDialog
- [ ] DropdownSelector in SearchView
- [ ] RichTextToolbar (mail editor, signature editor, event editor, out of office notification editor)

File: src/gui/base/RichTextToolbar.ts [hunk 3]
```diff
@@ -23,6 +23,14 @@ export interface RichTextToolbarAttrs {
 export class RichTextToolbar implements Component<RichTextToolbarAttrs> {
 	selectedSize = size.font_size_base
 
+	constructor({attrs}: Vnode<RichTextToolbarAttrs>) {
+		try {
+			this.selectedSize = parseInt(attrs.editor.squire.getFontInfo().size.slice(0, -2))
```
[the remark this answers is not in the document]
Comment (johnbotris) [reply src/gui/base/RichTextToolbar.ts:28]: it gets updated in the onclick of the size button

https://github.com/tutao/tutanota/blob/637bf2408f00bf7c99fededdeef5f1c0e4ddd8fd/src/gui/base/RichTextToolbar.ts#L210

[the remark this answers is not in the document]
Comment (charlag) [reply src/gui/base/RichTextToolbar.ts:28]: Yeah I rather meant that it won't change because of the attributes change but I think it's fine

**Pull request merged**